src/kata6.joinNames.js

```diff
@@ -1,3 +1,25 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+  let nameText = "";
+  let outputString = "";
 
+  for (let i = 0; i < namesObj.length; i++) {
+    nameText = namesObj[i].name;
+
+    if (i < namesObj.length - 2) {
+      outputString += nameText;
+      outputString += ", ";
+    }
+
+    if (i === namesObj.length - 2) {
+      outputString += nameText;
+    }
+
+    if (i === namesObj.length - 1) {
+      outputString += " & ";
+      outputString += nameText;
+    }
+  }
```

Comment:
So theres a couple of things that can be done here to make this a bit cleaner. 
- the number of variables in the global scope can be reduced, as `nameText` is only referenced inside the for loop,
- We can declare `nameText` inside the for loop as it will be reassigned on each iteration,
- the separate additions to thee `outputString` inside the first and third if statements can be rolled into one using a template literal

```
  let outputString = "";

  for (let i = 0; i < namesObj.length; i++) {
    const nameText = namesObj[i].name;

    if (i < namesObj.length - 2) {
      outputString += `${nameText}, `;
    }

    if (i === namesObj.length - 2) {
      outputString += nameText;
    }

    if (i === namesObj.length - 1) {
      outputString += ` & ${nameText}`;
    }
  }

  return outputString;
};
```

Hope that makes sense!